src/test/ensemble_config_change_stress.cpp

```diff
@@ -461,43 +462,128 @@ class StressPipelineConfigChanges : public TestWithTempDir {
 
         for (auto& [retCode, counter] : createPipelineRetCodesCounters) {
             SPDLOG_TRACE("Create:[{}]={} -- {}", static_cast<uint>(retCode), counter, ovms::Status(retCode).string());
-            if (requiredCreatePDResults.find(retCode) != requiredCreatePDResults.end()) {
+            if (requiredLoadResults.find(retCode) != requiredLoadResults.end()) {
                 EXPECT_GT(counter, 0) << static_cast<uint>(retCode) << ":" << ovms::Status(retCode).string() << " did not occur. This may indicate fail or fail in test setup";
                 continue;
             }
             if (counter == 0) {
                 continue;
             }
-            EXPECT_TRUE(allowedCreatePDResults.find(retCode) != allowedCreatePDResults.end()) << "Ret code:"
-                                                                                              << static_cast<uint>(retCode) << " message: " << ovms::Status(retCode).string()
-                                                                                              << " was not allowed in test but occured during load";
+            EXPECT_TRUE(allowedLoadResults.find(retCode) != allowedLoadResults.end()) << "Ret code:"
+                                                                                      << static_cast<uint>(retCode) << " message: " << ovms::Status(retCode).string()
+                                                                                      << " was not allowed in test but occured during load";
         }
-        for (auto& [retCode, counter] : executePipelineRetCodesCounters) {
-            SPDLOG_TRACE("Execute:[{}]={} -- {}", static_cast<uint>(retCode), counter, ovms::Status(retCode).string());
-            if (requiredExecuteResults.find(retCode) != requiredExecuteResults.end()) {
-                EXPECT_GT(counter, 0) << static_cast<uint>(retCode) << ":\"" << ovms::Status(retCode).string() << "\" did not occur. This may indicate fail or fail in test setup";
-                continue;
+    }
+    bool isMetadataResponseCorrect(tensorflow::serving::GetModelMetadataResponse& response) {
+        tensorflow::serving::SignatureDefMap def;
+        EXPECT_EQ(response.model_spec().name(), pipelineName);
+        EXPECT_TRUE(response.model_spec().has_version());
+        EXPECT_EQ(response.model_spec().version().value(), 1);
+        EXPECT_EQ(response.metadata_size(), 1);
+        EXPECT_NE(
```

Comment:
Here we have EXPECT. We should not proceed with the test any futher if we are missing `signature_def`. It would mean that unpacking the signature will fail at line 486/487 (probably with segfault).

Reply:
Yes you are right that we should exit early in such case. I added for the same reason boolean values few lines later on.

However you cannot use ASSERT inside functions non returning void & it does not make you test exit early if used in void returning function.